spec.emu

```diff
@@ -67,9 +67,11 @@ contributors: Chengzhong Wu, Hemanth HM, Ecma International
             1. Let _msg_ be ? ToString(_message_).
             1. Let _msgDesc_ be the PropertyDescriptor { [[Value]]: _msg_, [[Writable]]: *true*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.
             1. Perform ! DefinePropertyOrThrow(_O_, `"message"`, _msgDesc_).
-          1. <ins>If _cause_ is not *undefined*, then</ins>
-            1. <ins>Let _causeDesc_ be the PropertyDescriptor { [[Value]]: _cause_, [[Writable]]: *true*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.</ins>
-            1. <ins>Perform ! DefinePropertyOrThrow(_O_, `"cause"`, _causeDesc_).</ins>
+          1. <ins>If Type(_options_) is Object, then</ins>
+            1. <ins>If ? HasProperty(_options_, `"cause"`) is *true*, then</ins>
+              1. <ins>Let _cause_ be ? Get(_options_, `"cause"`)</ins>
+              1. <ins>Let _causeDesc_ be the PropertyDescriptor { [[Value]]: _cause_, [[Writable]]: *true*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.</ins>
+              1. <ins>Perform ! DefinePropertyOrThrow(_O_, `"cause"`, _causeDesc_).</ins>
```

Comment:
```suggestion
              1. <ins>Perform ! CreateDataPropertyOrThrow(_O_, `"cause"`, _cause_).</ins>
```

Reply:
`CreateDataPropertyOrThrow` creates different descriptors from the `message` descriptors: the data property descriptors are `{ [[Value]]: V, [[Writable]]: true, [[Enumerable]]: true, [[Configurable]]: true }`, the `error.message` property descriptors are `{ [[Value]]: msg, [[Writable]]: true, [[Enumerable]]: false, [[Configurable]]: true }.`. The key difference is if they are enumerable. I believe it would be better if we keep consistent on the property descriptors here?

Reply:
Ah, fair point. Disregard this one then.